src/app/core/view/dynamic-components/dynamic-component.directive.ts

```diff
@@ -22,16 +25,23 @@ export class DynamicComponentDirective implements OnChanges {
     private registry: ViewRegistry
   ) {}
 
-  ngOnChanges(): void {
-    this.loadDynamicComponent();
+  async ngOnChanges() {
+    await this.loadDynamicComponent();
   }
 
-  private loadDynamicComponent() {
+  private async loadDynamicComponent() {
     if (!this.appDynamicComponent) {
       return;
     }
 
-    const component = this.registry.get(this.appDynamicComponent.component);
+    let component: ComponentType<OnInitDynamicComponent>;
+    if (this.registry.has(this.appDynamicComponent.component)) {
+      component = this.registry.get(this.appDynamicComponent.component);
+    } else {
+      component = await dynamicComponents.get(
+        this.appDynamicComponent.component
+      )();
+    }
```

Comment:
if (almost) everything is lazy-loaded, do we stil have use for the registry at all?